kv/replica_slice.go

```diff
@@ -105,7 +107,9 @@ func (rs replicaSlice) SortByCommonAttributePrefix(attrs []string) int {
 	topIndex := len(rs) - 1
 	for bucket := 0; bucket < len(attrs); bucket++ {
 		firstNotOrdered := 0
+		rs.SortRandomOrder(firstNotOrdered)
```

Comment:
Modified the place calling SortRandomOrder inside loop when there are no matched bucket.